inst/ad_advs.R

```diff
@@ -0,0 +1,260 @@
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#derive_param)
+
+# Derive BMI
+advs <- advs |>
+  filter(VSTESTCD != "BMI") |>
+  derive_param_bmi(
+    by_vars = exprs(
+      STUDYID, USUBJID, TRTSDT, TRTEDT, TRT01P, TRT01A, VISIT,
+      VISITNUM, ADT, ADY, VSTPT, VSTPTNUM
+    ),
+    set_values_to = exprs(
+      PARAMCD = "BMI",
+      DOMAIN = "VS"
+    ),
+    get_unit_expr = VSSTRESU,
+    filter = VSSTAT != "NOT DONE" | is.na(VSSTAT),
+    constant_by_vars = exprs(USUBJID)
+  )
+
+
+# Derive categorization variables ----
+# See the "Derive Categorization Variables" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#cat)
+
+# Derive BMI class (AVALCAT1, AVALCA1N)
+advs <- advs |>
+  derive_vars_cat(
+    definition = avalcat_lookup,
+    by_vars = exprs(PARAMCD)
+  )
+
+
+# Derive Baseline variables ----
+# See the "Derive Baseline" and "Derive Change from Baseline " vignette sections
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#baseline)
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#bchange)
+
+# Add baseline flag (ABLFL)
+advs <- advs |>
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+      order = exprs(ADT, VSTPTNUM, VISITNUM),
+      new_var = ABLFL,
+      mode = "last"
+    ),
+    filter = (!is.na(AVAL) & ADT <= TRTSDT)
+  )
+
+# Derive baseline analysis value (BASE)
+advs <- advs |>
+  derive_var_base(
+    by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+    source_var = AVAL,
+    new_var = BASE
+  )
+
+# Derive absolute (CHG) and relative (PCHG) change from baseline
+advs <- advs |>
+  derive_var_chg()
+advs <- advs |>
+  derive_var_pchg()
+
+
+# Derive criterion variables ----
+# See the "Derive Criterion Variables" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#crit_vars)
+
+# Set weight loss criterion flags (CRIT1, CRIT1FL)
+advs <- advs |>
+  derive_vars_crit_flag(
+    condition = PCHG <= -5 & PARAMCD == "WEIGHT",
+    description = "Achievement of ≥ 5% weight reduction from baseline",
+    crit_nr = 1,
+    values_yn = TRUE,
+    create_numeric_flag = FALSE
+  ) |>
+  derive_vars_crit_flag(
+    condition = PCHG <= -10 & PARAMCD == "WEIGHT",
+    description = "Achievement of ≥ 10% weight reduction from baseline",
+    crit_nr = 2,
+    values_yn = TRUE,
+    create_numeric_flag = FALSE
+  )
+
+
+# Assign parameter variables ----
+# See the "Assign PARAMCD, PARAM, PARAMN, PARCAT1" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#paramcd)
+
+# Add all parameter variables (PARAM, PARAMN, PARCAT1)
+advs <- advs |>
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    new_vars = exprs(PARAMCD, PARAM, PARAMN, PARCAT1, PARCAT1N),
```

Comment:
Is `PARAM` required here as it's already in ADVS?

Reply:
It is not added during the early join (only PARAMCD is joined) as that is the way the Admiral vignette does it.

Reply:
You're right sorry, I must've mistaken this block for the one at the beginning where only `PARAMCD` is merged.